modules/core/src/main/scala/com/romanidze/kaesquella/core/client/ClientInterpreter.scala

```diff
@@ -0,0 +1,74 @@
+package com.romanidze.kaesquella.core.client
+
+import com.romanidze.kaesquella.core.models.{ClientError, KSQLVersionResponse, StatusInfo}
+import com.romanidze.kaesquella.core.models.ksql.{Request => KSQLInfoRequest}
+import com.romanidze.kaesquella.core.models.query.{Request => KSQLQueryRequest}
+import com.romanidze.kaesquella.core.models.query.row.RowInfo
+import com.romanidze.kaesquella.core.models.ksql.ddl.DDLInfo
+import com.romanidze.kaesquella.core.models.ksql.stream.StreamResponse
+import com.romanidze.kaesquella.core.models.ksql.table.TableResponse
+import com.romanidze.kaesquella.core.models.ksql.query.QueryResponse
+
+/**
+ * Trait for describing common client operations. May be changed!
+ *
+ * @author Andrey Romanov
+ * @since 0.0.1
+ * @version 0.0.1
+ */
+trait ClientInterpreter[F[_], G[_]] {
+
+  /**
+   * Method for retrieving information about query status
+   *
+   * @return query status
+   */
+  def getQueryStatus(queryID: String): F[Either[ClientError, StatusInfo]]
```

Comment:
Consider creating alias for Either[ClientError, A]. Now it's not very ergonomic

Reply:
Hi Bulat! :D
Yeah, another option would be to simplify `F[Either[Error, T]]` to just `F[T]` with error channel inside `MonadError[F, Throwable]`. Then you need to have Error as a subtype of Throwable, as described [here](https://nrinaudo.github.io/scala-best-practices/adts/errors_extend_exception.html), for example. Otherwise, signatures are very complex (especially in runQueryRequest method)